metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/service/TestDatasetService.java

```diff
@@ -109,47 +126,54 @@ public void testCreateDataset() throws Exception {
         datasetArgumentCaptor.getValue().getOrganizationId());
     assertEquals(metisUser.getOrganizationName(),
         datasetArgumentCaptor.getValue().getOrganizationName());
+    verify(authorizer, times(1)).authorizeWriteNewDataset(metisUser);
+    verifyNoMoreInteractions(authorizer);
   }
 
-  @Test(expected = UserUnauthorizedException.class)
```

Comment:
In general I like the new approach of expected exceptions but they should normally be used when some extra information of the exception should be checked (e.g. message) otherwise I find the annotation based check cleaner.

On the other hand if expected exceptions are used why not use the existing implementation of junit? It's very simple and straightforward I would say..
https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html

Same goes for every other test in this PR.

Reply:
One more thing. The ExpectedException does indeed not continue after catching the exception. So another approach is needed.

I did some more digging. It turns out that JUnit 5 has something that we could use (assertThrows(), works pretty much exactly like what I made), but we cannot use it as we are still at JUnit 4. Also, there is a google project that can do this (called Catch-Exception, https://code.google.com/archive/p/catch-exception/), but that would require a new dependency for something that is simple and will be included in a new version of JUnit. 

One thing to be careful of with @Test(expeced = ...) is the danger of false positives: if the setup code throws the exact same exception, this will be reported as a successful test. This is why I don't particularly like this option. But for now most of the code has it and until we upgrade to JUnit 5 we should probably keep it that way.

Anyway, I did not make any further changes.

Reply:
Yes indeed, as I said above, the annotation based approach should be avoided when there are extra information of the exception to be checked.
JUnit 5 will be good to upgrade soon then.
Ok let's keep it as is! Please push the changes.